### ISSUE TITLE
Cannot build AOT when the package System.Text.Json is referenced

### ISSUE THREAD

**Issue by Daoting** [unoplatform/Uno.Wasm.Bootstrap#255]
`WasmShellMonoRuntimeExecutionMode `is  InterpreterAndAOT, when the package System.Text.Json is referenced, it will generate an error. It’s not uno's fault, but I don’t know who to bring the problem to. 

```
linker-out/Uno.Wasm.StaticLinking.pdb linker-out/System.Text.Json.dll linker-out/System.Text.Encodings.Web.dll linker-out/System.Numerics.Vectors.dll linker-out/System.Numerics.Vectors.pdb linker-out/System.Threading.Tasks.Extensions.dll linker-out/Microsoft.Bcl.AsyncInterfaces.dll linker-out/System.Buffers.dll linker-out/System.Runtime.CompilerServices.Unsafe.dllFatal error in Mono IL Linker

Unhandled Exception:System.NotImplementedException: Initialization of return value in method 'System.ReadOnlySpan`1<System.UInt32> System.Text.Unicode.UnicodeHelpers::GetDefinedCharacterBitmap()'  at Mono.Linker.Steps.RemoveUnreachableBlocksStep+BodySweeper.GetReturnInitialization (Mono.Cecil.Cil.Instruction[]& initInstructions) [0x000e2] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Steps.RemoveUnreachableBlocksStep+BodySweeper.Process (System.Collections.Generic.List`1[T] conditionInstrsToRemove) [0x00046] in <650915bc2d964f7c92b8aa7bb4fb0096>:0  
 at Mono.Linker.Steps.RemoveUnreachableBlocksStep+BodyReducer.RewriteBody () [0x0006d] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   at Mono.Linker.Steps.RemoveUnreachableBlocksStep.RewriteBody (Mono.Cecil.MethodDefinition method) [0x0001e] in <650915bc2d964f7c92b8aa7bb4fb0096>:0 
  at Mono.Linker.Steps.RemoveUnreachableBlocksStep.RewriteBodies (Mono.Collections.Generic.Collection`1[T] types) [0x0005b] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   at Mono.Linker.Steps.RemoveUnreachableBlocksStep.Process () [0x0008a] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Steps.BaseStep.Process (Mono.Linker.LinkContext context) [0x00010] in <650915bc2d964f7c92b8aa7bb4fb0096>:0 
  at Mono.Linker.Pipeline.ProcessStep (Mono.Linker.LinkContext context, Mono.Linker.Steps.IStep step) [0x0000d] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Pipeline.Process (Mono.Linker.LinkContext context) [0x0000f] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Driver.Run (Mono.Linker.ILogger customLogger) [0x0193e] in <650915bc2d964f7c92b8aa7bb4fb0096>:0  
 at Mono.Linker.Driver.Execute (System.String[] args, Mono.Linker.ILogger customLogger) [0x0004b] in <650915bc2d964f7c92b8aa7bb4fb0096>:0  
 at Mono.Linker.Driver.Main (System.String[] args) [0x00000] in <650915bc2d964f7c92b8aa7bb4fb0096>:0 [ERROR] FATAL 

UNHANDLED EXCEPTION: System.NotImplementedException: Initialization of return value in method 'System.ReadOnlySpan`1<System.UInt32> System.Text.Unicode.UnicodeHelpers::GetDefinedCharacterBitmap()'  at Mono.Linker.Steps.RemoveUnreachableBlocksStep+BodySweeper.GetReturnInitialization (Mono.Cecil.Cil.Instruction[]& initInstructions) [0x000e2] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Steps.RemoveUnreachableBlocksStep+BodySweeper.Process (System.Collections.Generic.List`1[T] conditionInstrsToRemove) [0x00046] in <650915bc2d964f7c92b8aa7bb4fb0096>:0  
 at Mono.Linker.Steps.RemoveUnreachableBlocksStep+BodyReducer.RewriteBody () [0x0006d] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   at Mono.Linker.Steps.RemoveUnreachableBlocksStep.RewriteBody (Mono.Cecil.MethodDefinition method) [0x0001e] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Steps.RemoveUnreachableBlocksStep.RewriteBodies (Mono.Collections.Generic.Collection`1[T] types) [0x0005b] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Steps.RemoveUnreachableBlocksStep.Process () [0x0008a] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
at Mono.Linker.Steps.BaseStep.Process (Mono.Linker.LinkContext context) [0x00010] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   
t Mono.Linker.Pipeline.ProcessStep (Mono.Linker.LinkContext context, Mono.Linker.Steps.IStep step) [0x0000d] in <650915bc2d964f7c92b8aa7bb4fb0096>:0  
 at Mono.Linker.Pipeline.Process (Mono.Linker.LinkContext context) [0x0000f] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   at Mono.Linker.Driver.Run (Mono.Linker.ILogger customLogger) [0x0193e] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   at Mono.Linker.Driver.Execute (System.String[] args, Mono.Linker.ILogger customLogger) [0x0004b] in <650915bc2d964f7c92b8aa7bb4fb0096>:0   at Mono.Linker.Driver.Main (System.String[] args) [0x00000] in <650915bc2d964f7c92b8aa7bb4fb0096>:0 [74/211] "/mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdk/emsdk" construct_env emsdk_env.shAdding directories to PATH:PATH += /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdkPATH += /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdk/upstream/emscriptenPATH += /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdk/node/12.18.1_64bit/binSetting environment variables:EMSDK = /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdkEM_CONFIG = /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdk/.emscriptenEM_CACHE = /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdk/upstream/emscripten/cacheEMSDK_NODE = /mnt/d/dt/client/Common/wasm/emsdk-1.39.11/emsdk/node/12.18.1_64bit/bin/nodeninja: build stopped: subcommand failed.
```

**Comment by jeromelaban** [comment]
Thanks for the report. This looks like a mono linker issue.

Could you try validating with the latest 1.4-dev package ?

**Comment by Daoting** [comment]
Thanks. There are still errors after upgrading.

**Comment by jeromelaban** [comment]
Looks like it'll have to be a mono issue. Could you create a small repro ?

**Comment by m-bonaldo** [comment]
Same problem here when`WasmShellMonoRuntimeExecutionMode` is InterpreterAndAOT or FullAOT.
I tried using the latest 1.5.0-dev.57 package and the updated mono-sdk from CI (https://jenkins.mono-project.com/job/test-mono-mainline-wasm/label=ubuntu-1804-amd64/5415/).
To reproduce it just reference System.Text.Json in an InterpreterAndAOT or FullAOT project (eg. Uno.Wasm.StaticLinking.Aot) and add something like `var json = new System.Text.Json.JsonException();` in the Main in order for the linker to process the assembly. If I add the assembly to the LinkerConfig.xml the error is different, but still it does not work.

**Comment by jeromelaban** [comment]
This issue seems to be related to the way code is generated by the compiler, and how the mono linker processes it. It's likely that the ILLinker from .NET 5 may not have this issue.

This will have to be revisited once .NET 5/6 support is added (#187)

**Comment by stevozilik** [comment]
I'm experiencing same issue when both Microsoft.AspNetCore.SignalR.Client 3.1.6 and Uno.SkiaSharp.Wasm 2.80.0-uno.483 are referenced in one WASM project. Note there is no direct reference to System.Text.Json

 [ERROR] FATAL UNHANDLED EXCEPTION: System.NotImplementedException: Initialization of return value in method 'System.ReadOnlySpan`1<System.UInt32> System.Text.Unicode.UnicodeHelpers::GetDefinedCharacterBitmap()'

**Comment by jeromelaban** [comment]
Fixed by #288
